### PR TITLE
Format created of transaction with date time value if the created date is the current date

### PR DESCRIPTION
**Pull request by dukenv0307**
### Fixed Issues

$ https://github.com/Expensify/App/issues/33814
PROPOSAL: https://github.com/Expensify/App/issues/33814#issuecomment-1873849469

### Tests

1. Create a money request with the date as the default value
2. Inspect the network and verify that the created in the payload of API and the created of transaction in the response is formatted with the format `YYYY-MM-DD HH:mm:ss`
- [x] Verify that no errors appear in the JS console

### Offline tests

1. Create a money request with the date as the default value
2. Check the transaction in OnyxDB and verify that the created of transaction formatted with the format `YYYY-MM-DD HH:mm:ss`
### QA Steps

1. Create a money request with the date as the default value
2. Inspect the network and verify that the created in the payload of API and the created of transaction in the response is formatted with the format `YYYY-MM-DD HH:mm:ss`
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/129500732/f7748402-3a7f-4fe9-aa41-e977b266671e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/129500732/1a188885-7b06-4b88-acf5-a7a7aae4a993

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/129500732/89e1917f-b3ce-4170-ae46-7e771246822d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/129500732/d2141a9d-4153-45c6-b9b8-6fb11e1dbb0e

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/129500732/ff0f535c-4357-4d87-8c8b-91b7f7c8fbf9

</details>

### CODE REVIEW

File: src/libs/actions/IOU.js [hunk 1]
```diff
@@ -870,6 +870,8 @@ function createDistanceRequest(report, participant, comment, created, category,
     // If the report is an iou or expense report, we should get the linked chat report to be passed to the getMoneyRequestInformation function
     const isMoneyRequestReport = ReportUtils.isMoneyRequestReport(report);
     const currentChatReport = isMoneyRequestReport ? ReportUtils.getReport(report.chatReportID) : report;
+    const currentTime = DateUtils.getDBTime();
+    const currentCreated = created === format(new Date(currentTime), CONST.DATE.FNS_FORMAT_STRING) ? currentTime : created;
```
Comment (cubuspl42) [inline src/libs/actions/IOU.js:874]: Isn't this equivalent to [isSameDay](https://date-fns.org/v3.2.0/docs/isSameDay)?

File: src/libs/actions/IOU.js [hunk 2]
```diff
@@ -870,6 +870,8 @@ function createDistanceRequest(report, participant, comment, created, category,
     // If the report is an iou or expense report, we should get the linked chat report to be passed to the getMoneyRequestInformation function
     const isMoneyRequestReport = ReportUtils.isMoneyRequestReport(report);
     const currentChatReport = isMoneyRequestReport ? ReportUtils.getReport(report.chatReportID) : report;
+    const currentTime = DateUtils.getDBTime();
```
Comment (cubuspl42) [inline src/libs/actions/IOU.js:873]: Please extract this logic to a function like...

```ts
/**
 * Conditionally add the time-of-day information to the request creation timestamp
 */
fun enrichMoneyRequestTimestamp(created) {
  // ...
  return currentCreated;
}
```

**Review by jakub-trzebiatowski — CHANGES_REQUESTED** [review]

File: src/libs/DateUtils.ts [hunk 3]
```diff
@@ -730,6 +730,15 @@ function formatToSupportedTimezone(timezoneInput: Timezone): Timezone {
     };
 }
 
+/**
+ * Return the date with full format if the created date is the current date.
+ * Otherwise return the created date.
+ */
+function enrichMoneyRequestTimestamp(created: string) {
+    const currentTime = getDBTime();
+    return isSameDay(new Date(created), new Date(currentTime)) ? currentTime : created;
```
Comment (cubuspl42) [inline src/libs/DateUtils.ts:739]: Please test if it works when dropping the `new Date(...)` wrapping call

File: src/libs/DateUtils.ts [hunk 4]
```diff
@@ -730,6 +730,15 @@ function formatToSupportedTimezone(timezoneInput: Timezone): Timezone {
     };
 }
 
+/**
+ * Return the date with full format if the created date is the current date.
+ * Otherwise return the created date.
+ */
+function enrichMoneyRequestTimestamp(created: string) {
```
Comment (cubuspl42) [inline src/libs/DateUtils.ts:737]: Please state the return type explicitly

File: src/libs/DateUtils.ts (hunk 3 above)
Comment (cubuspl42) [reply src/libs/DateUtils.ts:739]: Oh, it does on `2.x` indeed. I [started a thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1705662293667419).

Comment (cubuspl42) [reply src/libs/DateUtils.ts:739]: There's some active discussion, but it won't be actionable for this PR. We're good here.

**Comment by cubuspl42** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/2590174/3cd1e70f-2302-49f9-b1ac-fa7aeedb80fb

</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/2590174/c56f6dd3-ee99-4065-af36-fa680de733c8

</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/2590174/9dd994fb-a1e2-44c9-95e7-7dd209a5044f

</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by jakub-trzebiatowski — APPROVED** [review]

**Comment by deetergp** [comment]
I'm not seeing the created date on the transaction containing a full datetime stamp.

Screenshot 2024-01-22 at 12 48 13 PM

**Comment by cubuspl42** [comment]
@deetergp This is definitely a problem! 😬

Was our testing approach incorrect?

Sorry if it's a trivial question, but why are you analyzing the response, not the request?

**Comment by cubuspl42** [comment]
@dukenv0307 Please test the PR by [faking the timezone](https://stackoverflow.com/a/60008052/1483676) and time so that locally, the date is different from that in UTC+0.

**Comment by deetergp** [comment]
Oh, I think I understand what's going on. Are both of you in UTC? The issue for me is that `isSameDay` is returning false because the DB is in UTC, but my browser is in PST, so `isSameDay` is returning false:

Screenshot 2024-01-23 at 12 12 11 PM

**Comment by cubuspl42** [comment]
@deetergp I'm UTC+1; I think @dukenv0307 is close.

There's no such think as distinguished `Date`/`DateTime`  in JS, I think; it's always the `Date` class, which _represents_ the datetime (i.e. date + time-of-day + timezone marker). When you parse `2023-01-01` to `Date`, you'll get the instant of UTC midnight for that date **in your local timezone**.

This is a giant mess and that's why I voted to go with Yoda Time instead of `date-fns`. Well, I was in the minority.

File: src/libs/DateUtils.ts [hunk 5]
```diff
@@ -730,6 +730,15 @@ function formatToSupportedTimezone(timezoneInput: Timezone): Timezone {
     };
 }
 
+/**
+ * Return the date with full format if the created date is the current date.
+ * Otherwise return the created date.
+ */
+function enrichMoneyRequestTimestamp(created: string): string {
+    const currentTime = getDBTime();
+    return formatWithUTCTimeZone(created) === formatWithUTCTimeZone(currentTime) ? currentTime : created;
```
Comment (cubuspl42) [inline src/libs/DateUtils.ts:739]: Sorry, I won't happily approve more smelly approaches to date operations. Let's stick with the best thing we got, using `date-fns` as much as possible.

You'll have to convince me _why_ that's not the good idea.

Comment (cubuspl42) [reply src/libs/DateUtils.ts:739]: Also, is this even correct? Aren't we ultimately forced to use the "DB time" format?

Comment (cubuspl42) [reply src/libs/DateUtils.ts:739]: I think that we should first ask ourselves what timezone is the original date-only `created` string. As I understand it, it's in the user-local timezone! So we should compare it with the user-local "now".

>  I think isToday function is helpful for us in this case.

Also what I started with, but likely it's not a clean solution. It, likely, will have a problem of time passing between two invocations of `now()` (`new Date()`).

Comment (cubuspl42) [reply src/libs/DateUtils.ts:739]: I sketched the best solution I could think of [here](https://github.com/Expensify/App/pull/34784#discussion_r1463029210). Let me know if it's unclear or if you disagree with the approach.

Comment (cubuspl42) [reply src/libs/DateUtils.ts:739]: I'm confused, those strings don't have a timezone marker. How does the backend know what is the timezone context of a string like "YYYY-MM-DD HH:MM:SS"?

Comment (cubuspl42) [reply src/libs/DateUtils.ts:739]: We shouldn't perform date/time algorithms on the string level, unless absolutely necessary. Especially with more code like "let's replace Ts and Zs with `' '` and we'll be fine". Let's not make this code worse than it already is.
 
Please explain to me what's wrong with this approach:

```ts
const now = new Date();
return isSameDate(new Date(created), now) ? getDBTimeFromDate(now) : created;
```

Comment (cubuspl42) [reply src/libs/DateUtils.ts:739]: Okey, let's try...

```ts
const now = new Date();
const createdDate = parse(created, CONST.DATE.FNS_FORMAT_STRING, now);
return isSameDate(createdDate, now) ? getDBTimeFromDate(now) : created;
```

File: src/libs/DateUtils.ts [hunk 6]
```diff
@@ -730,6 +730,23 @@ function formatToSupportedTimezone(timezoneInput: Timezone): Timezone {
     };
 }
 
+/**
+ * Returns the time in milliseconds of a date in the format expected by the database
```
Comment (cubuspl42) [inline src/libs/DateUtils.ts:734]: Time in milliseconds?

File: src/libs/DateUtils.ts [hunk 7]
```diff
@@ -730,6 +730,23 @@ function formatToSupportedTimezone(timezoneInput: Timezone): Timezone {
...
+ */
+function getDBTimeFromDate(date: Date): string {
```
Comment (cubuspl42) [inline src/libs/DateUtils.ts:736]: Well, please use it inside `getDBTime` now

File: src/libs/DateUtils.ts (hunk 5 above)
Comment (deetergp) [reply src/libs/DateUtils.ts]: Time zones are the worst. Just to clarify, all datetimes that get saved to our back end should be stored as UTC strings in the `YYYY-MM-DD HH:MM:SS` format. But the kicker is that when displaying them back to the user, we probably need to show it relative to their time zone.

Comment (cubuspl42) [reply src/libs/DateUtils.ts]: Timezones are the worst, true, but we're not helping ourselves by making very questionable decisions like passing dates around as strings, converting `Date`s to string to `Date`s to strings to `Date`..., not using separate types for instants and dates and datetimes, etc., etc.

File: src/libs/DateUtils.ts [hunk 8]
```diff
@@ -360,12 +360,19 @@ function getMicroseconds(): number {
     return Date.now() * CONST.MICROSECONDS_PER_MS;
 }
 
+/**
+ * Returns the format yyyy-MM-dd HH:mm:ss of a date in the format expected by the database
```
Comment (cubuspl42) [inline src/libs/DateUtils.ts:364]: Okey, let's nuke this comment (as it's a helper function related to `getDBTime`) and update the original comment so it makes sense.

File: src/libs/DateUtils.ts [hunk 9]
```diff
@@ -360,12 +360,19 @@ function getMicroseconds(): number {
...
+ */
+function getDBTimeFromDate(date: Date): string {
+    return date.toISOString().replace('T', ' ').replace('Z', '');
+}
+
 /**
  * Returns the current time in milliseconds in the format expected by the database
```
Comment (cubuspl42) [inline src/libs/DateUtils.ts:371]:
```ts
/**
 * Convert the given timestamp to the "yyyy-MM-dd HH:mm:ss" format, as expected by the database 
 *
 * @param [timestamp] the given timestamp (if omitted, defaults to the current time)
 */
```

**Review by jakub-trzebiatowski — CHANGES_REQUESTED** [review]

File: src/libs/DateUtils.ts (hunk 8 above)
Comment (cubuspl42) [reply src/libs/DateUtils.ts:364]: 👉 https://github.com/Expensify/App/pull/34784#discussion_r1466183362

**Review by jakub-trzebiatowski — APPROVED** [review]

**Comment by cubuspl42** [comment]
I tested this with timezone spoofing and I believe the logic is timezone-proof now!

@deetergp Would you confirm it fixed the problem for you?

**Comment by deetergp** [comment]
@cubuspl42 Yep, it fixed the issue for me! The money request I made created a transaction in the data base with UTC timestamp of when I created it, in the expected format.

**Review by deetergp — APPROVED** [review]
Looks good and tests well, thanks for the changes.

**Pull request merged**